cli/dotfiles.go

```diff
@@ -246,11 +269,60 @@ func (r *RootCmd) dotfiles() *clibase.Cmd {
 			Description: "Specifies the directory for the dotfiles symlink destinations. If empty, will use $HOME.",
 			Value:       clibase.StringOf(&symlinkDir),
 		},
+		{
+			Flag:          "branch",
+			FlagShorthand: "b",
+			Description: "Specifies which branch to clone. " +
+				"If empty, will default to cloning the default branch or using the existing branch in the cloned repo on disk.",
+			Value: clibase.StringOf(&gitbranch),
+		},
 		cliui.SkipPromptOption(),
 	}
 	return cmd
 }
 
+type ensureCorrectGitBranchParams struct {
+	repoDir       string
+	gitSSHCommand string
+	gitBranch     string
+}
+
+func ensureCorrectGitBranch(baseInv *clibase.Invocation, params ensureCorrectGitBranchParams) error {
+	dotfileCmd := func(cmd string, args ...string) *exec.Cmd {
+		c := exec.CommandContext(baseInv.Context(), cmd, args...)
+		c.Dir = params.repoDir
+		c.Env = append(baseInv.Environ.ToOS(), fmt.Sprintf(`GIT_SSH_COMMAND=%s -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no`, params.gitSSHCommand))
+		var out bytes.Buffer
```

Comment:
Out created multiple times? Perhaps use io.Discard below as a default?

Reply:
Whoops this was a copy/paste mistake. That should be baseInv.StdOut by default.